src/unix/osd.c

```diff
@@ -66,5 +74,104 @@ int fi_wait_cond(pthread_cond_t *cond, pthread_mutex_t *mut, int timeout)
 	return pthread_cond_timedwait(cond, mut, &ts);
 }
 
+int ofi_shm_map(struct util_shm *shm, const char *name, int size,
+		int readonly, void **mapped)
+{
+	char *fname = 0;
+	int ret = FI_SUCCESS;
+	int flags = O_RDWR | (readonly ? 0 : O_CREAT);
+	struct stat mapstat;
+
+	int i;
+
+	*mapped = MAP_FAILED;
+	memset(shm, 0, sizeof(*shm));
+
+	fname = (char*)calloc(1, strlen(name) + 2); /* '/' + %s + trailing 0 */
+	if (!fname) {
+		ret = -FI_ENOMEM;
+		goto failed;
+	}
+	strcpy(fname, "/");
+	strcat(fname, name);
+	shm->name = fname;
+
+	for (i = 0; i < strlen(fname); i++) {
+		if (fname[i] == ' ')
+			fname[i] = '_';
+	}
+
+	FI_DBG(&core_prov, FI_LOG_CORE,
+		"Creating shm segment :%s (size: %lu)\n", fname, size);
```

Comment:
here was another issue: size_t was declared as int instead of size_t. now it is fixed (and %lu is correct)

thank you